x/orderbook/msg.go

```diff
@@ -0,0 +1,104 @@
+package orderbook
+
+import (
+	"github.com/iov-one/weave"
+	coin "github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/migration"
+)
+
+func init() {
+	// Migration needs to be registered for every message introduced in the codec.
+	// This is the convention to message versioning.
+	migration.MustRegister(1, &CreateOrderBookMsg{}, migration.NoModification)
+	migration.MustRegister(1, &CreateOrderMsg{}, migration.NoModification)
+	migration.MustRegister(1, &CancelOrderMsg{}, migration.NoModification)
```

Comment:
```NoModification is a migration function that migrates data that requires no change. It should be used to register migrations that do not require any modyfications.``` This is what is written on [weave/migration/register.go](../migration/register.go) but I think this needs more explanation. On which cases to use ```migration.NoModification``` and ```migration.RefuseMigration```?

Reply:
For registering the schema 1, you should always have `migration.NoModification` to initialize it.

For future schemas, they can be a soft change (with a migration path), or a hard break (no migration path). Hard schema changes in models may be next to impossible to implement on a chain without a strong manual hard-fork.